### PR TITLE
[Feature]Enable Custom Font Color

### PR DESCRIPTION
**Pull request by inancyGit**
## Purpose

* ...

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull request checklist

This PR has considered:
- [x] Color Theming
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] View Data Injection
- [ ] Demo App UI/UX update

## How to Test

* Open...
* Click on...

## What to Check

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | Screenshot or description with this change |

### CODE REVIEW

**Review by jamchengms — DISMISSED** [review]

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Utilities/ThemeColor.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import Foundation
+import UIKit
+import FluentUI
+
+struct ThemeColor: ThemeOptions {
+    init() {
```
Comment (ShaunaSong) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Utilities/ThemeColor.swift:11]: Why we need empty init() method here?

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/PushNotificationEventType.swift [hunk 2]
```diff
@@ -77,6 +77,8 @@ extension AzureCommunicationCalling.PushNotificationEventType {
             return .incomingPstnCall
         case .stopRinging:
             return .stopRinging
```
Comment (ShaunaSong) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/PushNotificationEventType.swift:79]:
```
@unknown default:
            fatalError("Error")
```
            
 Is this code change related to theming feature?

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/ThemeOptions.swift [hunk 3]
```diff
@@ -47,4 +51,9 @@ public extension ThemeOptions {
     var primaryColorTint30: UIColor {
         return Colors.Palette.communicationBlueTint30.color
     }
+    /* <CUSTOM_COLOR_FEATURE> */
+    var foregroundOnPrimaryColor: UIColor {
+        return .white
```
Comment (pavelprystinka) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/ThemeOptions.swift:56]: Should we use a color palette here?

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/PushNotificationEventType.swift (hunk 2 above)
Comment (inancyGit) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/PushNotificationEventType.swift:79]: No, it was for pipeline warning

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/ThemeOptions.swift (hunk 3 above)
Comment (inancyGit) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/ThemeOptions.swift:56]: yes, this uses a public extension and asking mandatorily to use like other themeOptions

**Review by ShaunaSong — DISMISSED** [review]

**Review by Mohtasim — DISMISSED** [review]

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/ThemeOptions.swift (hunk 3 above)
Comment (pavelprystinka) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/ThemeOptions.swift:56]: But why do we use UIColor.white and in the rest of the options we use Colors.Palette.communicationBlue.color?

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/SwiftUI/ViewComponents/Button/PrimaryButton.swift [hunk 4]
```diff
@@ -9,20 +9,20 @@ struct PrimaryButton: View {
...
                         iconName: viewModel.iconName,
-                        paddings: viewModel.paddings) {
+                        paddings: viewModel.paddings,
+                        themeOptions: viewModel.themeOptions) {
             $0.addTarget(action, action: #selector(Action.perform(sender:)), for: .touchUpInside)
             action.action = {
                 viewModel.action()
             }
         }
-            .onTapGesture(perform: viewModel.action)
```
Comment (pavelprystinka) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/SwiftUI/ViewComponents/Button/PrimaryButton.swift:24]: Why do we delete .onTapGesture(perform: viewModel.action)
It is still needed for iOS < 18
$0.addTarget(action, action: #selector(Action.perform(sender:)), for: .touchUpInside) is added to work on iOS18 recently

**Review by pavelprystinka — CHANGES_REQUESTED** [review]

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/ThemeOptions.swift (hunk 3 above)
Comment (inancyGit) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/ThemeOptions.swift:56]: Because its foreground color,  focusing on button title, kept a white as default.

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/SwiftUI/ViewComponents/Button/PrimaryButton.swift (hunk 4 above)
Comment (inancyGit) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/SwiftUI/ViewComponents/Button/PrimaryButton.swift:24]: @pavelprystinka , I just updated the one with ThemeOptions. Not made any changes to  .onTapGesture(perform: viewModel.action).

Comment (inancyGit) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/SwiftUI/ViewComponents/Button/PrimaryButton.swift:24]: It came from the previous code of develop. Once I merge it, it will be fixed.

**Review by pavelprystinka — APPROVED** [review]

**Review by ShaunaSong — APPROVED** [review]
